### PR TITLE
Allows for wall mirror frames to be crafted

### LINKED ISSUE #8316: Newscaster frames attach incorrectly.
Issue reported from  Round ID: 42252 (BeeStation Sage - Roleplay)

Reporting client version: 514.1583

image

### PR DESCRIPTION
**Pull request by CydiaLamiales**
## About The Pull Request
Adds a new craftable item - wall mirror frame.
* closes https://github.com/BeeStation/BeeStation-Hornet/issues/8316

## Why It's Good For The Game
More buildable structures good.

## Testing Photographs and Procedure

<details>
<summary>Screenshots&Videos</summary>

Crafting and placing a wall mirror :

https://user-images.githubusercontent.com/110184118/216757728-8018ca07-02e2-4d02-8eec-3816c2aeb99b.mp4

Handheld wall mirror frame icon :

image

All the placeable objects that had their code changed : 
image

</details>

## Changelog
:cl:
add: Added a new craftable iteml - wall mirror frame
imageadd: Added an icon for the wall mirror frame
code: Made the wall painting and photo frames and the fire extinguisher cabinet use the proper code for wall frame pixel shifting
fix: Fixed the Newscaster being placed in the opposite direction when using a Newscaster wall frame
/:cl:

### CODE REVIEW

File: code/game/objects/structures/mirror.dm [hunk 1]
```diff
@@ -12,10 +12,17 @@
...
 	. = ..()
 	if(icon_state == "mirror_broke" && !broken)
 		obj_break(null, mapload)
+	if(mapload)
+		return
+	if(dir)
+		setDir(dir)
+	if(building)
+		pixel_x = (dir & 3)? 0 : (dir == 4 ? -30 : 30)
+		pixel_y = (dir & 3)? (dir ==1 ? -30 : 30) : 0
```
Comment (PowerfulBacon) [inline code/game/objects/structures/mirror.dm]: Use the defines for direction rather than magic numbers

**Review by PowerfulBacon — DISMISSED** [review]

File: code/game/objects/structures/mirror.dm (hunk 1 above)
Comment (CydiaLamiales) [reply code/game/objects/structures/mirror.dm]: Oh alright, this was copy and pasted from code for other wall frames.

File: code/game/objects/structures/mirror.dm [hunk 2]
```diff
@@ -11,8 +11,12 @@
 	flags_ricochet = RICOCHET_SHINY
 	layer = ABOVE_WINDOW_LAYER
 	var/magical = FALSE
+	var/dir_north = NORTH
+	var/dir_east = EAST
+	var/dir_south = SOUTH
+	var/dir_west = WEST
```
Comment (itsmeow) [inline code/game/objects/structures/mirror.dm]: why

File: code/game/objects/structures/extinguisher.dm [hunk 3]
```diff
@@ -14,9 +14,6 @@
 /obj/structure/extinguisher_cabinet/Initialize(mapload, ndir, building)
 	. = ..()
 	if(building)
-		setDir(ndir)
-		pixel_x = (dir & 3)? 0 : (dir == 4 ? -27 : 27)
-		pixel_y = (dir & 3)? (dir ==1 ? -30 : 30) : 0
```
Comment (itsmeow) [inline code/game/objects/structures/extinguisher.dm:19]: what does removing this do even

**Review by itsmeow — CHANGES_REQUESTED** [review]

File: code/game/objects/structures/extinguisher.dm (hunk 3 above)
Comment (CydiaLamiales) [reply code/game/objects/structures/extinguisher.dm:19]: It pixel shifts the wall structure to actually be on the wall when placed. It's unnecessary however, as the wall frame item already has a variable that can handle that.

File: code/game/objects/structures/mirror.dm (hunk 2 above)
Comment (CydiaLamiales) [reply code/game/objects/structures/mirror.dm]: Oh guh, that was leftovers from testing stuff, forgot to delete

File: code/game/objects/structures/mirror.dm [hunk 4]
```diff
@@ -10,9 +10,8 @@
 	integrity_failure = 100
 	flags_ricochet = RICOCHET_SHINY
 	layer = ABOVE_WINDOW_LAYER
-	var/magical = FALSE
```
Comment (itsmeow) [inline code/game/objects/structures/mirror.dm:13]: you made an oopsie

Comment (CydiaLamiales) [reply code/game/objects/structures/mirror.dm:13]: crap

**Review by itsmeow — APPROVED** [review]

**Pull request merged**